yaml_edit/lib/src/map_mutations.dart

```diff
@@ -47,19 +44,16 @@ SourceEdit updateInMap(
 
 /// Performs the string operation on [yaml] to achieve the effect of removing
 /// the element at [key] when re-parsed.
-SourceEdit removeInMap(YamlEditor yamlEdit, YamlMap map, Object key) {
-  ArgumentError.checkNotNull(yamlEdit, 'yamlEdit');
-  ArgumentError.checkNotNull(map, 'map');
-
+SourceEdit? removeInMap(YamlEditor yamlEdit, YamlMap map, Object? key) {
```

Comment:
How can this return `null` ?

Reply:
This is the body for `removeInMap()`:

```dart
SourceEdit? removeInMap(YamlEditor yamlEdit, YamlMap map, Object? key) {
  if (!containsKey(map, key)) return null;
...
```
...so I had to change the return value to nullable on the 2nd pass of changes I did without checking how it would be used. After changing the test to force a fail for this scenario, it would seem that the `containsKey()` call is no longer necessary since it's covered by `_traverse()`;

```sh
00:01 +1 -1: throws PathError if collectionPath is invalid [E]                                                                                                          
  Invalid path: [d]. Failed to traverse to subpath (d)! Parent {a: 1, b: 2, c: 3} does not contain key or index d.
  package:yaml_edit/src/editor.dart 502:25  YamlEditor._pathErrorOrElse
  package:yaml_edit/src/editor.dart 479:18  YamlEditor._traverse
  package:yaml_edit/src/editor.dart 413:26  YamlEditor.remove
  test/remove_test.dart 39:11               main.<fn>.<fn>
  
00:01 +1 -2: throws PathError if collectionPath is invalid in nested path [E]                                                                                           
  Invalid path: [a, c]. Failed to traverse to subpath (a, c)! Parent {b: foo} does not contain key or index c.
  package:yaml_edit/src/editor.dart 502:25  YamlEditor._pathErrorOrElse
  package:yaml_edit/src/editor.dart 479:18  YamlEditor._traverse
  package:yaml_edit/src/editor.dart 413:26  YamlEditor.remove
  test/remove_test.dart 49:11               main.<fn>.<fn>
```

Reply:
`removeInMap()` is only used in `remove()`.